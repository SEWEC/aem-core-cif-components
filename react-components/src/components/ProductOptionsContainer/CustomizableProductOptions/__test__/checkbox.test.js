/*******************************************************************************
 *
 *    Copyright 2020 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import Checkbox from '../checkbox';
import { fireEvent } from '@testing-library/react';
import { render } from 'test-utils';

describe('<Checkbox>', () => {
    const option_id = 1;
    const required = true;

    const sortedOptions = [
        {
            id: 1,
            price: 12,
            label: 'Carmina Necklace'
        },
        {
            id: 2,
            price: 13,
            label: 'Augusta Necklace'
        }
    ];

    const customization = [
        {
            id: 1,
            price: 12,
            label: 'Carmina Necklace'
        }
    ];

    const handleSelectionChange = jest.fn();

    it('renders the component', () => {
        const { asFragment } = render(
            <Checkbox
                option_id={option_id}
                required={required}
                currencyCode="USD"
                options={sortedOptions}
                customization={customization}
                handleSelectionChange={handleSelectionChange}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('tests selection change', () => {
        const { getByRole } = render(
            <Checkbox
                option_id={option_id}
                required={required}
                currencyCode="USD"
                options={sortedOptions}
                customization={customization}
                handleSelectionChange={handleSelectionChange}
            />
        );

        fireEvent.click(getByRole('checkbox', { name: 'Augusta Necklace + $13.00' }));

        const newCustomization = [
            {
                id: 1,
                price: 12,
                label: 'Carmina Necklace'
            },
            {
                id: 2,
                price: 13,
                label: 'Augusta Necklace'
            }
        ];

        expect(handleSelectionChange).toHaveBeenCalledTimes(1);
        expect(handleSelectionChange).toHaveBeenCalledWith(option_id, newCustomization);
    });
});
