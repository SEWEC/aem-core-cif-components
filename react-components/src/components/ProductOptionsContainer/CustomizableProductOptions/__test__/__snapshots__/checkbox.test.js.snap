// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox> renders the component 1`] = `
<DocumentFragment>
  <div
    class="product__option"
  >
    <label>
      <input
        checked=""
        type="checkbox"
        value="1"
      />
       Carmina Necklace +
      <b>
        <span
          class=""
        >
          $12.00
        </span>
      </b>
    </label>
  </div>
  <div
    class="product__option"
  >
    <label>
      <input
        type="checkbox"
        value="2"
      />
       Augusta Necklace +
      <b>
        <span
          class=""
        >
          $13.00
        </span>
      </b>
    </label>
  </div>
</DocumentFragment>
`;
